packages/devtools_app/lib/src/performance/performance_model.dart

```diff
@@ -441,9 +441,23 @@ class FlutterFrame {
     type ??= event?.type;
     if (type == TimelineEventType.ui) {
       time.start = event?.time?.start;
+      if (rasterEventFlow != null) {
+        time.end = event?.time?.end;
+      }
     }
     if (type == TimelineEventType.raster) {
-      time.end = event?.time?.end;
+      // If there is not a [uiEventFlow] for this frame, there will either never
```

Comment:
nit: this comment is a little hard to read due to be written as almost a double negative.

Reply:
fixed.